src/Orleans.Persistence.Migration/Serialization/IGrainStorageSerializer.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans.Runtime;
+
+namespace Orleans.Storage
+{
+    /// <summary>
+    /// Common interface for grain state serializers.
+    /// </summary>
+    public interface IGrainStorageSerializer
+    {
+        /// <summary>
+        /// Serializes the object input.
+        /// </summary>
+        /// <param name="input">The object to serialize.</param>
+        /// <typeparam name="T">The input type.</typeparam>
+        /// <returns>The serialized input.</returns>
+        BinaryData Serialize<T>(T input);
+
+        /// <summary>
+        /// Deserializes the provided data.
+        /// </summary>
+        /// <param name="input">The data to deserialize.</param>
+        /// <typeparam name="T">The output type.</typeparam>
+        /// <returns>The deserialized object.</returns>
+        T Deserialize<T>(BinaryData input);
```

Comment:
I wonder if we need to solve the issue of textual vs binary data here. Eg, for JSON-based storage systems like Postgres (json columns), Mongo DB, & CosmosDB